tokio/src/net/driver/reactor/mod.rs

```diff
@@ -386,13 +393,13 @@ impl Inner {
             .get_readiness(token)
             .unwrap_or_else(|| panic!("token {} no longer valid!", token));
 
-        let (waker, ready) = match dir {
-            Direction::Read => (&sched.reader, !mio::Ready::writable()),
-            Direction::Write => (&sched.writer, mio::Ready::writable()),
+        let (waker, interest) = match dir {
+            Direction::Read => (&sched.reader, Readiness::READABLE),
+            Direction::Write => (&sched.writer, Readiness::WRITABLE),
         };
 
         waker.register(w);
-        if readiness & ready.as_usize() != 0 {
+        if readiness & interest.as_usize() != 0 {
             waker.wake();
```

Comment:
This call is now quite a bit more expansive then in Mio v0.6, since in v0.7 this is a system call. You might want to consider using a different mechanism for this.